lib/Optimizer/Transforms/GenDeviceCodeLoader.cpp

```diff
@@ -91,22 +91,27 @@ class GenerateDeviceCodeLoader
         // called by our cudaq kernel
         // Set of dependent kernels that we've included.
         DenseSet<llvm::StringRef> includedNeededFuncs;
-        funcOp.walk([&](func::CallOp callOp) {
-          if (auto neededFunc =
-                  module.lookupSymbol<func::FuncOp>(callOp.getCallee())) {
-            if (neededFunc.empty())
-              return WalkResult::skip();
-            // If we've included this kernel, e.g. it's called a second time,
-            // skip.
-            if (includedNeededFuncs.contains(neededFunc.getName()))
-              return WalkResult::skip();
-            neededFunc.print(strOut, opf);
-            includedNeededFuncs.insert(neededFunc.getName());
-            strOut << '\n';
-            return WalkResult::advance();
-          }
-          return WalkResult::advance();
-        });
+        const std::function<WalkResult(func::CallOp)> emitDependentFunc =
```

Comment:
This variable isn't really needed.

Reply:
I'm curious ... if this variable isn't needed, how would you perform the recursive call in line 108?

Reply:
We're already walking every function in the module. This just turns a linear scan into a recursive dive back through the same functions. Why do that?

Create a map for each function in turn. Then scan that map and create the transitive closure.

Reply:
Thanks @schweitzpgi. Now I see your point. I added #1366 to get it fixed.